torchreid/losses/asl.py

```diff
@@ -118,3 +122,18 @@ def forward(self, cos_theta, targets, scale=None, aug_index=None, lam=None):
             self.loss *= one_sided_w
 
         return - self.loss.sum()
+
+
+def label_smooth(targets, smooth_degree):
+    if smooth_degree > 0:
+        targets = targets * (1 - smooth_degree)
+        targets[targets == 0] = smooth_degree
+    return targets
+
+
+def filter_input(inputs, targets):
+    valid_idx = targets >= 0
```

Comment:
In which cases targets could be < 0 and why it is not handled during dataset initialization or `__getitem__()` method?

Reply:
In the future task could generate such outputs if some labels are ignored. We can filter negative elements in getitem, but collate function expects all targets to have the same length.